packages/adapter-xata/src/xata.ts

```diff
@@ -0,0 +1,88 @@
+import { buildClient, BaseClientOptions, XataRecord, ClientConstructor } from "@xata.io/client";
+
+export interface NextauthUser {
```

Comment:
These types already exist. You can import them from

```ts
import type { AdapterUser, VerificationToken } from "next-auth/adapters"
import type { Account } from "next-auth"
```

(We should probably re-export `Account` from `next-auth/adapters` as well. :eyes:)

If you need to augment these types in your own project, check out https://next-auth.js.org/getting-started/typescript#module-augmentation

Reply:
Right, this file is automatically generated from a Xata database though. We don't really write this by hand. 😬

Reply:
Do you mean from the schema? Should this be part of the adapter then or something the user generates for their project? There are fields here (ie. `location`) that are not part of the default User model. https://next-auth.js.org/adapters/models/

Reply:
Great catch! Thanks a lot for this! Fixing...

Reply:
> Do you mean from the schema? Should this be part of the adapter then, or something the user generates for their project? There are fields here (ie. location) that are not part of the default User model. https://next-auth.js.org/adapters/models/
> Related: https://github.com/nextauthjs/next-auth/pull/4911/files#r919547972

So, we're using the type `XataClient` from this file, which has a list of tables we need. These types are generated from the schema, correct. When the user imports their generated client, it will have to be a superset of this one to be type-safe (meaning, it'll have to have our required tables).

Thanks for catching the flags not on the user model. I've removed them.

Reply:
I'll resolve this now 🙌